### ISSUE TITLE
Async call in onFileUploadStart

### ISSUE THREAD

**Issue by jordonias** [expressjs/multer#78]
Currently there exists the ability to stop an upload by returning false.
```javascript
onFileUploadStart: function (file) {
  if (file.originalname == 'virus.exe') return false;
}
```
I'm suggesting the ability to provide a function with two arguments where the second argument is a callback. If `cb(false)` is called the file is not uploaded.

```javascript
onFileUploadStart: function (file, cb) {
  somethingAsync(function(err) {
    if(err) return cb(false);
    // Do something
    return cb();
  });
}
```

There may be a better away of approaching this, or a reason this wouldn't work. Input is appreciated. Otherwise I can work on submitting a PR.

**Comment by jpfluger** [comment]
onFileUploadStart does now include `req` and `res`, the relevant line in code being 107.

`var proceed = options.onFileUploadStart(file, req, res);`

Maybe you can work with `req` or `res` on your issue? What were you trying to solve? (I'm guessing a friendly response back to the client? ... :+1: )

**Comment by LinusU** [comment]
This is fixed in the newly released version `1.0.0`. Use the `fileFilter` option which gives you a callback to call :+1:
